### DISCUSSION TITLE
How can I add an environment variable that begins with a double underscore?

### DISCUSSION THREAD

**Discussion by mbrameld** [vercel/community#2854] [Help]
### Summary

I have an issue that is worked around by adding an environment variable like `__NEXT_PRIVATE_PREBUNDLED_REACT=next`. When I try to add that to my Vercel project, I get an error message saying that environment variables have to start with a letter. How can I add this environment variable to my Vercel project?

### Example

https://github.com/vercel/next.js/issues/48964

### Steps to Reproduce

1. Try to add an environment variable named `__NEXT_PRIVATE_PREBUNDLED_REACT` to a Vercel project.

**Comment by smaeda-ks** [comment]
@mbrameld as a workaround, you should be able to set it in your `.env` file separately. I'll see if we can support that on UI as well

**Reply by mbrameld** [reply]
It's set in my .env file, everything works fine running locally. The issue is when I try to deploy to Vercel. Unless you mean I can somehow edit a .env file through the Vercel UI? I'm not going to check .env files into source control.

**Reply by smaeda-ks** [reply]
@mbrameld As long as `.env` (or `.env.production`) is not added to `.gitignore`/`.vercelignore`, they will be automatically loaded upon your next build:
https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#environment-variable-load-order

You can have both `.env` file and Vercel env vars (added via UI), and they will be merged accordingly. If the same env var key (name) is found, Vercel env var will take precedence.

**Reply by smaeda-ks** [reply]
so a possible workaround is to use `.env.local` instead of `.env` for local dev, and define `__NEXT_PRIVATE_PREBUNDLED_REACT ` in `.env` or `.env.production` and track them in your source control.
https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#default-environment-variables
